ddtrace/tracer/option.go

```diff
@@ -415,6 +412,14 @@ func (c *config) loadAgentFeatures() {
 	}
 }
 
+func (c *config) canComputeStats() bool {
+	return c.agent.Stats && c.HasFeature("discovery")
```

Comment:
Given this get used in the span finish method, WDYT of caching this computed value could be worth to avoid the underlying map lookup everytime.

Reply:
I thought about that but didn't include it to keep the code simpler (don't want to prematurely optimize).

I can do some benchmarks to see what kind of overhead the map lookup has and if it's worth caching.

Reply:
I was thinking of a simple:
```go
if cachedValue != nil { return cachedValue }
cachedValue = [...]
return cachedValue
```

Reply:
This is a benchmark just removing the map lookup to estimate the best possible improvement by caching:
It appears adding the map lookup only costs `~100ns` to the start/finish of a span. I'm not totally convinced it's that high, since `canComputeStats` and `canDropP0s` only gain .4ns. Maybe they're smaller due to data or instruction caching in the CPU 🤷 

I did change it from `HasFeature` to `HasFlag`, which skips the string trimming, which had a larger (although also small) impact. Anyway, I don't think 100ns is worth caching the value for.

```
% benchcmp no_map_lookup.bench with_has_flag.bench 
benchmark                      old ns/op     new ns/op     delta
BenchmarkStatsConfig-8         4659          4781          +2.62%
BenchmarkCanComputeStats-8     0.28          0.64          +131.95%
BenchmarkCanDropP0s-8          0.28          0.63          +119.82%

benchmark                      old allocs     new allocs     delta
BenchmarkStatsConfig-8         28             28             +0.00%
BenchmarkCanComputeStats-8     0              0              +0.00%
BenchmarkCanDropP0s-8          0              0              +0.00%

benchmark                      old bytes     new bytes     delta
BenchmarkStatsConfig-8         2847          2822          -0.88%
BenchmarkCanComputeStats-8     0             0             +0.00%
BenchmarkCanDropP0s-8          0             0             +0.00%
```

Reply:
Ok, nice check. Feel free to close this conversation if you're good :-)